### ISSUE TITLE
Automated support for generating API documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3069]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3069|
|Type|enhancement|
|Created|2008-03-08 21:15:27Z|
|Branch|https://github.com/twisted/twisted/tree/api-builder-3069-3|

Create a kind of builder in `twisted.python._release` which emits pydoctor-generated API docs.

<details><summary>Searchable metadata</summary>

```
trac-id__3069 3069
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__totally_automated_release_infrastructure totally automated release infrastructure
branch__branches_api_builder_3069_3 branches/api-builder-3069-3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1205010927000000 1205010927000000
changetime__1207064063000000 1207064063000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [22778]) Branching to 'api-builder-3069'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [22806]) Branching to 'api-builder-3069-2'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

 * please pass the '--quiet' option to pydoctor
 * test should be skipped if pydoctor is not importable

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [23145]) Branching to 'api-builder-3069-3'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner|
|-|-|

Done.  Note this still requires a branch of pydoctor, it won't work with trunk (that was true last time too but since I didn't mention it before...).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve set owner to @exarkun|
|-|-|

OK, I have finally found what was wrong: you're not passing the system-class option. That explains why the private members are generated with the `APIBuilder` class.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun removed owner|
|-|-|

That's a pretty important thing I guess!  Thanks.  The code specifies a system class now.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set owner to @exarkun|
|-|-|

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve commented|
|-|-|

For some reasons, the 2.3 builder tries to import pydoctor but then fails: http://buildbot.twistedmatrix.com/builders/debian-py2.3-select/builds/2582/steps/trial/logs/problems.  We probably want to fix the slave.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [23166]) Merge api-builder-3069-3

Author: exarkun, itamarst
Reviewer: therve
Fixes [#3069](https://github.com/twisted/twisted/issues/3069)

Add a release automation class for generating API documentation
using pydoctor.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
